logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/MessageContent.kt

```diff
@@ -26,8 +26,9 @@ sealed class MessageContent {
 
     data class DeleteForMe(
         val messageId: String,
-        val conversationId: String,
-        val qualifiedConversationId: QualifiedConversationId?
+        @Deprecated("Use qualified id instead", ReplaceWith("conversationId"))
+        val unqualifiedConversationId: String,
```

Comment:
I don't understand why we add the `unqualifiedConversationId` and immediately mark it as deprecated. Wouldn't it be easier to inject the self domain into the `ProtobufContentMapper` and let it convert unqualified IDs into qualified ones.

Reply:
Hehehe, that was one of the solutions Mateusz and I thought about.

The main reason to go this way was to follow the same approach done for `HideMessage`.

But we can break this pattern now.

I suggest one improvement more:

We could just stop using `UserRepository` everywhere just to get self user ID. We could inject `selfUserId` directly into the constructor, because the `UserSessionScope` already has a `userId` anyway when we create it.


This way we could inject a `defaultDomain` or similar, that would help solving this Mapper thing.

Reply:
Sounds like a plan! If you already discussed this among yourself I'm fine with doing this refactoring in a separate PR.

Reply:
I think tackeling EventReceiver etc in a seperate refactor would be a good idea to include this issue as well so I am going to leave as it is for now